src/Common/MessageHandler.ts

```diff
@@ -22,7 +22,7 @@ export function handleCachedDataMessage(message: any): void {
   if (messageContent.error != null) {
     cachedDataPromise.deferred.reject(messageContent.error);
   } else {
-    cachedDataPromise.deferred.resolve(JSON.parse(messageContent.data));
```

Comment:
Note that `handleCachedDataMessage()` isn't used anywhere, so I'm modifying the behavior to return the message content data as is, rather than parsing it.